File: playground2/user/repository/user.repository.js
const {users} = require('../model/user.model');

function findUserByEmail (email) {
    return users.find((u) => u.email === email);
}

function createUser(email, hashedPassword) {
    const user = {email, password: hashedPassword};
    users.push(user);
    return user;
}

module.exports = {
    findUserByEmail,
    createUser
};

File: playground3/src/main/java/com/example/playground3/repository/UserRepository.java
package com.example.playground3.repository;

import com.example.playground3.model.User;
import org.springframework.stereotype.Repository;

import java.util.ArrayList;
import java.util.List;

@Repository
public class UserRepository {

    private final List<User> users = new ArrayList<>();

    public User findUserByEmail(String email) {
        return users.stream()
                .filter(u -> u.getEmail().equals(email))
                .findFirst()
                .orElse(null);
    }

    public User createUser(String email, String hashedPassword) {
        User user = new User(email, hashedPassword);
        users.add(user);
        return user;
    }
}
